hardware/chisel/src/test/scala/unittest/TensorGemmJsonTest.scala

```diff
@@ -0,0 +1,278 @@
+      }
+      if (peek(c.io.out.wr(0).valid) == 1) {
+        expect(c.io.out.wr(0).bits.idx, out_indices.dequeue())
+      }
+    }
+
+    def test_if_done() {
+      println(s"uop_indices should be empty ${uop_indices.size}")
+      println(s"acc_indices should be empty ${acc_indices.size}")
+      println(s"inp_indices should be empty ${inp_indices.size}")
+      println(s"wgt_indices should be empty ${wgt_indices.size}")
+      println(s"accout_indices should be empty ${accout_indices.size}")
+      println(s"out_indices should be empty ${out_indices.size}")
+    }
+
+    def check() = {
+      val result = for {
+        ((x,y),idx) <- (acc_scratchpad, acc_o_scratchpad).zipped.toList.zipWithIndex
+      } yield {
+        (for {((xx,yy),jdx) <- (x,y).zipped.toList.zipWithIndex} yield {
+          if (xx != yy) {
+            println(s"Value mismatch at $idx $jdx: $xx (actual) != $yy (expected)")
+          }
+          xx == yy
+        }).reduce((x,y) => x&&y)
+      }
+      val result2 = result.reduce((x,y) => x&&y)
+      result2
+    }
+  }
+
+  val mocks = new Mocks
+
+  for {
+    cnt_o <- BigInt(0) until lp_0
+    cnt_i <- BigInt(0) until lp_1
+    uop_idx <- uop_begin until uop_end
+  } {
+    val u0 = uop_scratchpad(uop_idx.toInt)(0)
+    val u1 = uop_scratchpad(uop_idx.toInt)(1)
+    val u2 = uop_scratchpad(uop_idx.toInt)(2)
+
+    mocks.uop_indices.enqueue(uop_idx)
+    mocks.acc_indices.enqueue(u0 + acc_0*cnt_o + acc_1*cnt_i)
+    mocks.inp_indices.enqueue(u1 + inp_0*cnt_o + inp_1*cnt_i)
+    mocks.wgt_indices.enqueue(u2 + wgt_0*cnt_o + wgt_1*cnt_i)
+    mocks.accout_indices.enqueue(u0 + acc_0*cnt_o + acc_1*cnt_i)
+
+    if (dec_reset == 0) {
+      mocks.out_indices.enqueue(u0 + acc_0*cnt_o + acc_1*cnt_i)
+    }
+  }
+
+  poke(c.io.start, 0)
+  mocks.logical_step()
+  expect(c.io.state, c.sIdle)
+  poke(c.io.start, 1)
+
+  val total_steps = (uop_end-uop_begin)*lp_0*lp_1
+
+  val max_count = 100 + 4*total_steps
+  var count = 0
+  while (peek(c.io.done) == 0 && count < max_count) {
+    if (count % 100 == 0) {
+      println(s"logical_step $count")
+    }
+    mocks.logical_step()
+    if (count == 0) {
+      poke(c.io.start, 0)
+    }
+    count += 1
+  }
+
+  assert(peek(c.io.done) == 1, s"Signal done never high even after $count steps.")
+  println(s"Signal done high after $count steps.")
+
+  mocks.logical_step()
+  expect(c.io.done, 0)
+
+  val cc = mocks.check()
+  println(s"Checking acc with acc_o ${cc}")
+  assert(cc)
+
+  println(s"Total active steps: ${total_steps}")
+  mocks.test_if_done()
+}
+
+class TensorGemmJsonTestx extends GenericTest("TensorGemmJson", (p:Parameters) => new TensorGemmPipelinedSplit()(p),
+  (c:TensorGemmPipelinedSplit) => new TensorGemmJsonTester(c, "/x.json"))
+
+class TensorGemmJsonTesty extends GenericTest("TensorGemmJson", (p:Parameters) => new TensorGemmPipelinedSplit()(p),
+  (c:TensorGemmPipelinedSplit) => new TensorGemmJsonTester(c, "/y.json"))
```

Comment:
Hey @adavare , nice PR.

can we change `x.json` and `y.json` to have more meaningful names? It looks like these are test-data...is there some information that `x` tests versus `y` that we could use for naming?

Reply:
Thanks for the suggestion, I've renamed both files.